pkg/lib/cockpit-upload-helper.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+function debug(...args: unknown[]) {
+    if (window.debugging == 'all' || window.debugging?.includes('upload'))
+        console.debug('debug[upload]', ...args);
+}
+
+class UploadError extends Error {
+    name = 'UploadError';
+    constructor(problem: string, message: cockpit.JsonObject) {
+        // XXX message.message is untranslated.  Should we just always use cockpit.message(problem)?
```

Comment:
Yes please -- we use `cockpit.message()` everywhere else, so if there's something wrong with it, then let's fix that. It usually expects the error *object*, and already returns its argument's `.message` field if present. Can you please adjust the API accordingly to receive a single error instead of split problem/message fields?